pkg/datasources/roles.go

```diff
@@ -0,0 +1,80 @@
+package datasources
+
+import (
+	"database/sql"
+	"github.com/Snowflake-Labs/terraform-provider-snowflake/pkg/snowflake"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"log"
+)
+
+var rolesSchema = map[string]*schema.Schema{
+	"roles": {
+		Type:        schema.TypeList,
+		Computed:    true,
+		Description: "All roles in the account, including built in roles",
```

Comment:
description should be changed to better match documentation:
"List of all the roles which you can view across your entire account, including the system-defined roles and any custom roles that exist."